pkg/api/v1alpha1/kepler_internal_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: all internal types can depend on public types
+// e.g. kepler-internal.spec.exporter can reuse ExporterSpec because the API is
+// considered stable but not vice-versa.
+
+type InternalExporterDeploymentSpec struct {
+	ExporterDeploymentSpec `json:",inline"`
+	// Image of kepler-exporter to be deployed
+	// +kubebuilder:validation:MinLength=3
+	Image string `json:"image"`
+
+	// Namespace where kepler-exporter will be deployed
+	// +kubebuilder:validation:MinLength=1
+	Namespace string `json:"namespace"`
+}
+
+type InternalExporterSpec struct {
+	// +kubebuilder:validation:Required
+	Deployment InternalExporterDeploymentSpec `json:"deployment"`
+}
+
+type DashboardSpec struct {
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type OpenShiftSpec struct {
+	// +kubebuilder:default=false
+	Enabled   bool          `json:"enabled,omitempty"`
+	Dashboard DashboardSpec `json:"dashboard,omitempty"`
```

Comment:
for openshift, can we keep the `Enabled` default value to true?

Reply:
Sure ... Internal API can break at any time :D